dapps/tezosDomains.tsx

```diff
@@ -217,6 +408,407 @@ export function tezosDomains(transactions: Array<transaction>): CustomView {
+                      </li>
+                      <li>Token id: {data.token_id.toString()}</li>
+                    </ul>
+                  ),
+                  price,
+                },
+              ];
+            }
+            case "place_offer": {
+              const data = placeOfferSchema.Execute(micheline);
+
+              return [
+                {
+                  action: "Place offer",
+                  description: (
+                    <ul className="list-inside list-disc space-y-1 pt-1 font-light">
+                      <li>
+                        Token contract: <Alias address={data.token_contract} />
+                      </li>
+                      <li>Token id: {data.token_id.toString()}</li>
+                      <li>Price: {mutezToTez(data.price.toNumber())} Tez</li>
+                      {!!data.expiration ? (
+                        <li>
+                          Expiration date:{" "}
+                          {new Date(data.expiration).toLocaleString()}
+                        </li>
+                      ) : null}
+                    </ul>
+                  ),
+                  price,
+                },
+              ];
+            }
+            case "remove_offer": {
+              const data = removeOfferSchema.Execute(micheline);
+
+              return [
+                {
+                  action: "Remove offer",
+                  description: (
+                    <ul className="list-inside list-disc space-y-1 pt-1 font-light">
+                      <li>
+                        Token contract: <Alias address={data.token_contract} />
+                      </li>
+                      <li>Token id: {data.token_id.toString()}</li>
+                    </ul>
+                  ),
+                  price,
+                },
+              ];
+            }
+            default:
+              return [];
+          }
+        }
+        case TOKEN_CONTRACT.ghostnet:
+        case TOKEN_CONTRACT.mainnet: {
+          switch (transaction.entrypoints) {
+            case "transfer": {
+              const data = tokenTransferSchema.Execute(micheline) as Array<{
+                from_: string;
+                txs: Array<{
+                  to_: string;
+                  token_id: BigNumber;
+                  amount: BigNumber;
+                }>;
+              }>;
+
+              return [
+                {
+                  action: "Token transfer",
+                  description: (
+                    <ul className="list-inside list-disc space-y-1 pt-1 font-light">
+                      {data.map(transfer => (
+                        <li>
+                          From: <Alias address={transfer.from_} />
+                          <ul
+                            className="list-inside list-disc"
+                            style={{ marginLeft: "3rem" }}
+                          >
+                            {transfer.txs.map(({ to_, token_id, amount }) => (
```

Comment:
As it's the domain is a token, it fits TZIP12, FA2 contract. The following is an example coping from taquito to get domain name back.
I'd like the following change:
- we can show domain instead token ID
- transfer.txs.length should always be 1. If it's not 1, it must some users mass up. because amount will always be 1 for each domain. domain name don't have copy. Therefore, here we can just show "from", "to", "domain name".

so as the other case's involve token ID.
 
```
// import { TezosToolkit, compose } from '@taquito/taquito';
// import { Tzip12Module, tzip12 } from "@taquito/tzip12";
// const Tezos = new TezosToolkit('rpc_url');

Tezos.addExtension(new Tzip12Module());

const contractAddress = "KT1REqKBXwULnmU6RpZxnRBUgcBmESnXhCWs";
const tokenId = 3397;

Tezos.contract.at(contractAddress, tzip12)
.then(contract => {
  println(`Fetching the token metadata for the token ID ${tokenId}...`);
  return contract.tzip12().getTokenMetadata(tokenId);
})
.then (tokenMetadata => {
  println(JSON.stringify(tokenMetadata, null, 2));
})
.catch(error => println(`Error: ${JSON.stringify(error, null, 2)}`));
```
the result:
Fetching the token metadata for the token ID 3397...
{
  "token_id": 3397,
  "decimals": 0,
  "name": "abc.gho",
  "symbol": "DOMAIN"
}